content/modules/r/pages/reading-and-writing-data.adoc

```diff
@@ -0,0 +1,135 @@
+= Reading and Writing Data
+
+== Examples
+
+=== How do I read a csv file called `grades.csv` into a data.frame?
+
+Note that the "." means the current working directory. So, if we were in "/home/john/projects", "./grades.csv" would be the same as "/home/john/projects/grades.csv". This is called a _relative_ path. Read this for a better understanding.
+
+[source, R]
+----
+dat <- read.csv("./grades.csv")
+head(dat)
+----
+
+[source, R]
+----
+   grade      year
+ 1   100    junior
+ 2    99 sophomore
+ 3    75 sophomore
+ 4    74 sophomore
+ 5    44    senior
+ 6    69    junior
+----
+
+=== How do I read a csv file called `grades.csv` into a data.frame using the function `fread`?
+
+Note: The function `fread` is part of the `data.table` package and which reads in dataset faster than `read.csv`. It is therefore recommended for reading in large datasets in R.
+
+[source, R]
+----
+r
+library(data.table)
+dat <- data.frame(fread("./grades.csv"))
+head(dat)
+----
+
+[source, R]
+----
+   grade      year
+ 1   100    junior
+ 2    99 sophomore
+ 3    75 sophomore
+ 4    74 sophomore
+ 5    44    senior
+ 6    69    junior
+----
+
+=== How do I read a csv file called `grades2.csv` where instead of being comma-separated, it is semi-colon-separated, into a data.frame?
+
+[source, R]
+----
+dat <- read.csv("./grades_semi.csv", sep=";")
+head(dat)
+----
+
+[source, R]
+----
+   grade      year
+ 1   100    junior
+ 2    99 sophomore
+ 3    75 sophomore
+ 4    74 sophomore
+ 5    44    senior
+ 6    69    junior
+----
+
+=== How do I prevent R from reading in strings as factors when using a function like `read.csv`?
+
+In R 4.0+, strings are not read in as factors, so you do not need to do anything special. For R < 4.0, use `stringsAsFactors`.
+
+[source, R]
+----
+dat <- read.csv("./grades.csv", stringsAsFactors=F)
+head(dat)
+----
+
+[source, R]
+----
+   grade      year
+ 1   100    junior
+ 2    99 sophomore
+ 3    75 sophomore
+ 4    74 sophomore
+ 5    44    senior
+ 6    69    junior
+----
+
+=== How do I specify the type of 1 or more columns when reading in a csv file?
+
+[source, R]
+----
+dat <- read.csv("./grades.csv", colClasses=c("grade"="character", "year"="factor"))
+str(dat)
+----
+
+[source, R]
+----
+'data.frame':    10 obs. of  2 variables:
+ $ grade: chr  "100" "99" "75" "74" ...
+ $ year : Factor w/ 4 levels "freshman","junior",..: 2 4 4 4 3 2 2 3 1 2
+----
+
+=== Given a list of csv files with the same columns, how can I read them in and combine them into a single dataframe?
```

Comment:
Given a list of csv files with the same columns, how can I read in and combine them into a one dataframe?